crates/compiler/can/src/expr.rs

```diff
@@ -1171,6 +1169,7 @@ pub fn canonicalize_expr<'a>(
                 output,
             )
         }
+        ast::Expr::TaskAwaitBang(..) => todo!("canonicalize_expr on TaskAwaitBang"),
```

Comment:
Does this need to be implemented? If this can't be hit, maybe replace that with an 'unreachable!'?